Modules/_multiprocessing/semaphore.c

```diff
@@ -271,9 +271,9 @@ sem_timedwait_save(sem_t *sem, struct timespec *deadline, PyThreadState *_save)
             delay = difference;
 
         /* sleep */
-        tvdelay.tv_sec = delay / 1000000;
-        tvdelay.tv_usec = delay % 1000000;
-        if (select(0, NULL, NULL, NULL, &tvdelay) < 0)
+        tsdelay.tv_sec = delay / 1000000;
+        tsdelay.tv_nsec = (delay % 1000000) * 1000;
+        if (nanosleep(&tsdelay, NULL) < 0)
             return MP_STANDARD_ERROR;
```

Comment:
I didn't know that this function uses a busy loop (sem_trywait + sleep). It has a bug: it uses system clock, rather than a monotonic clock. I would prefer to first rewrite it using _PyTime_t and _PyTime_GetMonotonic(), before changing it to use nanosleep().

Can you please revert this change?